src/Federation/Resolvers/Entities.php

```diff
@@ -32,11 +33,25 @@ public function __invoke($root, array $args, GraphQLContext $context, ResolveInf
     {
         $results = [];
 
+        /** @var array<string, array<int, array<string, mixed>>> $groupedRepresentations */
+        $groupedRepresentations = [];
         foreach ($args['representations'] as $representation) {
-            $typename = $representation['__typename'];
-            $resolver = $this->entityResolverProvider->resolver($typename);
+            $groupedRepresentations[$representation['__typename']][] = $representation;
+        }
 
-            $results[] = $resolver($representation);
+        foreach ($groupedRepresentations as $typename => $representations) {
+            assert(is_string($typename), 'Never numeric due to GraphQL\Utils::isValidNameError()');
+
+            $resolver = $this->entityResolverProvider->resolver($typename);
+            if ($resolver instanceof BatchedEntityResolver) {
+                foreach ($resolver($representations) as $result) {
+                    $results[] = $result;
+                }
+            } else {
+                foreach ($representations as $representation) {
+                    $results[] = $resolver($representation);
+                }
+            }
```

Comment:
Could we think of a way to make the resolver itself responsible for this instead, so the if else statement wouldn't be needed? 🤔

Reply:
This conditional maintains backwards compatibiltity with non-batched entity resolvers. I would not know how to solve this any other way, without forcing developers to change their code or write more then the bare minimum code required.

Also, by doing grouping and merging here, we could change the logic of ordering around without the developer having to do anything.